core/src/main/java/hudson/DNSMultiCast.java

```diff
@@ -77,16 +84,102 @@ public Object call() {
-//            } catch (final IOException e) {
-//                LOGGER.log(Level.WARNING,"Failed to close down JmDNS instance!",e);
-//            }
+            } catch (final IOException e) {
+                LOGGER.log(Level.WARNING,"Failed to close down JmDNS instance!",e);
+            }
         }
     }
 
     private static final Logger LOGGER = Logger.getLogger(DNSMultiCast.class.getName());
 
-    public static boolean disabled = SystemProperties.getBoolean(DNSMultiCast.class.getName()+".disabled");
+    public static boolean disabled = SystemProperties.getBoolean(DNSMultiCast.class.getName()+".disabled", true);
+
+    private static class JenkinsJmDNS extends JmDNSImpl {
+        private static Logger logger = Logger.getLogger(JmDNSImpl.class.getName());
+        private final Class parent;
+
+        /**
+         * Create an instance of JmDNS and bind it to a specific network interface given its IP-address.
+         *
+         * @param address IP address to bind to.
+         * @param name    name of the newly created JmDNS
+         * @throws IOException
+         */
+        public JenkinsJmDNS(InetAddress address, String name) throws IOException {
+            super(address, name);
+            this.parent = this.getClass().getSuperclass();
+        }
+
+        /**
+         * Works like {@link #close()} but terminate uncleanly, but rapidly and without blocking.
+         */
+        public void abort() throws IOException {
+            if (this.isClosing()) {
+                return;
+            }
+
+           if (logger.isLoggable(Level.FINER)) {
+                logger.finer("Aborting JmDNS: " + this);
+            }
+            // Stop JmDNS
+            // This protects against recursive calls
+            if (this.closeState()) {
+                // We got the tie break now clean up
+
+                // Stop the timer
+                logger.finer("Canceling the timer");
+                this.cancelTimer();
+
+                // Cancel all services
+                executePrivateParentMethod("disposeServiceCollectors");
+
+                // Stop the canceler timer
+                logger.finer("Canceling the state timer");
+                this.cancelStateTimer();
+
+                // Stop the executor
+                shutdown();
+
+                // close socket
+                executePrivateParentMethod("closeMulticastSocket");
+
+                // remove the shutdown hook
+                if (_shutdown != null) {
+                    Runtime.getRuntime().removeShutdownHook(_shutdown);
+                }
+
+                if (logger.isLoggable(Level.FINER)) {
+                    logger.finer("JmDNS closed.");
+                }
+            }
+            advanceState(null);
+        }
+
+        private void shutdown() throws IOException {
+            try {
+                Field executor = this.parent.getDeclaredField("_executor");
+                executor.setAccessible(true);
```

Comment:
It will lead to IllegalReflectoiveAccess warnings on Java 11. Would there be a way to avoid that in the new jmdns version?

Reply:
The methods and fields that are executed/accessed by reflection are private and they are not exposed in any way